pkg/windows/windows.go

```diff
@@ -157,13 +157,15 @@ type windows struct {
 	// 		 in vSphere
 	//		https://bugzilla.redhat.com/show_bug.cgi?id=1876987
 	platform oconfig.PlatformType
-	// hostName is the name of the Windows VM that we need to configure in vSphere clusters. This is currently not set
-	// in case of vSphere VMs. In case of Linux, ignition was handling it. As we don't have an equivalent of ignition
-	// in Windows, we are setting this in WMCO currently
+	// machineName is the name of the Machine to be configured. This will only be set if the source of the VM is a
+	// Machine. In the case of BYOH, this field will be empty. The name of the Machine must be the same as the
+	// hostname of the associated VM. This is currently not true in the case of vSphere VMs provisioned by MAPI.
+	// In case of Linux, ignition was handling it. As we don't have an equivalent of ignition in Windows, we are
+	// WMCO must correct this by changing the VM's hostname.
 	// TODO: Remove this once we figure out how to do this via guestInfo in vSphere
 	// 		https://bugzilla.redhat.com/show_bug.cgi?id=1876987
-	hostName string
-	log      logr.Logger
+	machineName string
```

Comment:
I prefer leaving this as `hostName` and stating that if `hostName` is not set it implies that WMCO does not have to explicitly set the host name for vSphere instances. It is possible we might need to set the host name even in cases where the provider is `None`.

I suggest thinking more about how we can abstract the existing `machineName` input i.e. `NodeConfig` should not have logic to figure out whether the request came from a `Machine` or `ConfigMap` or special casing for that matter.